This patch was created by Daniel Robbins (drobbins@funtoo.org) and provides a means
to easily disable syntax highlighting via Pygments in literal_blocks in Sphinx.

In Sphinx, syntax highlighting is 'always on' so that things like '#' characters
will always be considered comments in literal blocks. This patch doesn't affect
default behavior but provides a property called 'highlight' which can be set to
False to turn off this behavior if not desired. Turning 'highlighting off' means
'don't have Pygments modify the output in any way.'

diff -urN Sphinx-4.2.0-bak/sphinx/writers/html.py Sphinx-4.2.0/sphinx/writers/html.py
--- Sphinx-4.2.0-bak/sphinx/writers/html.py	2021-07-11 11:41:57.000000000 -0600
+++ Sphinx-4.2.0/sphinx/writers/html.py	2021-10-20 15:15:29.821249112 -0600
@@ -450,7 +450,8 @@
 
     # overwritten
     def visit_literal_block(self, node: Element) -> None:
-        if node.rawsource != node.astext():
+        highlight = node.get('highlight', True)
+        if not highlight or node.rawsource != node.astext():
             # most probably a parsed-literal block -- don't highlight
             return super().visit_literal_block(node)
 
diff -urN Sphinx-4.2.0-bak/sphinx/writers/html5.py Sphinx-4.2.0/sphinx/writers/html5.py
--- Sphinx-4.2.0-bak/sphinx/writers/html5.py	2021-08-29 01:02:58.000000000 -0600
+++ Sphinx-4.2.0/sphinx/writers/html5.py	2021-10-20 15:16:09.197742118 -0600
@@ -401,7 +401,8 @@
 
     # overwritten
     def visit_literal_block(self, node: Element) -> None:
-        if node.rawsource != node.astext():
+        highlight = node.get('highlight', True)
+        if not highlight or node.rawsource != node.astext():
             # most probably a parsed-literal block -- don't highlight
             return super().visit_literal_block(node)
 
diff -urN Sphinx-4.2.0-bak/sphinx/writers/latex.py Sphinx-4.2.0/sphinx/writers/latex.py
--- Sphinx-4.2.0-bak/sphinx/writers/latex.py	2021-09-12 01:52:53.000000000 -0600
+++ Sphinx-4.2.0/sphinx/writers/latex.py	2021-10-20 15:16:07.833725040 -0600
@@ -1778,7 +1778,8 @@
         pass
 
     def visit_literal_block(self, node: Element) -> None:
-        if node.rawsource != node.astext():
+        highlight = node.get('highlight', True)
+        if not highlight or node.rawsource != node.astext():
             # most probably a parsed-literal block -- don't highlight
             self.in_parsed_literal += 1
             self.body.append(r'\begin{sphinxalltt}' + CR)
